docs/src/DeveloperGuide/LogicSeparation.md

```diff
@@ -0,0 +1,493 @@
+high-level, the use of a GPU for simulation roughly follows this diagram:
+
+```text
+┌─────────────┐     Programming     ┌─────────────┐
+│             │  State preparation  │             │
+│             ├─────────────────────▶             │
+│             │                     │             │
+│   Primary   │                     │  Secondary  │
+│ processing  │                     │ processing  │
+│    unit     │                     │    unit     │
+│             │                     │             │
+│             ◀─────────────────────┤             │
+│             │   Result readout    │             │
+└─────────────┘                     └─────────────┘
+```
+
+First the secondary processing unit (SPU) is programmed and as a part of this
+process the initial state (i.e. registers and memory) is prepared. The SPU then
+executes the program and the results are read out by the primary processing unit
+(PPU).
+
+## Classical-quantum processing architecture
+
+While one could imagine a quantum architecture in which everything is integrated
+into a single processing unit that both can process classical and quantum
+registers, it is unlikely that this kind of architecture will emerge in the near
+future due to the vastly different environments in which the two processes take
+place. Therefore, one feasible architecture is one in which we consider the
+quantum processing unit (QPU) as SPU and the CPU the PPU much like is the case
+for CPUs and GPUs today.
+
+Disregarding the more advanced scenarios where PPUs and SPUs work in parallel,
+the basic flow of a classical-quantum program execution follows a pattern as
+given below:
+
+```text
+      Classical             PPU-SPU             Quantum
+   processing unit  │      interface      │ processing unit
+                    │                     │
+                  │ │                     │
+Classical program │ │                     │
+        execution │ │                     │
+                  │ │  State preparation  │
+                  ▼ │     Programming     │
+                    ├─────────────────────▶
+                    │                     │ │
+                    │                     │ │ Quantum program
+                    │                     │ │ execution
+                    │   Result readout    │ │
+                    ◀─────────────────────┤ ▼
+                  │ │                     │
+Classical program │ │                     │
+        execution │ │  State preparation  │
+                  │ │     Programming     │
+                  ▼ ├─────────────────────▶
+                    │                     │ │
+                    │                     │ │ Quantum program
+                    │                     │ │ execution
+                    │   Result readout    │ │
+                    ◀─────────────────────┤ ▼
+                    │                     │
+                    ▼                     ▼
+                    Time
+```
+
+TODO: Finish this section
```

Comment:
Do you need to address this TODO in this PR?

Reply:
Fixed.